src/main/java/org/graylog/integrations/okta/OktaService.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.integrations.okta;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Inject;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import org.graylog2.inputs.InputService;
+import org.graylog2.plugin.system.NodeId;
+import org.graylog2.shared.inputs.MessageInputFactory;
+
+import java.io.IOException;
+
+public class OktaService {
+
+    private final InputService inputService;
+    private final MessageInputFactory messageInputFactory;
+    private final NodeId nodeId;
+    private final ObjectMapper objectMapper;
+
+    @Inject
+    public OktaService(InputService inputService, MessageInputFactory messageInputFactory, NodeId nodeId,
+                       ObjectMapper objectMapper) {
+
+        this.inputService = inputService;
+        this.messageInputFactory = messageInputFactory;
+        this.nodeId = nodeId;
+        this.objectMapper = objectMapper;
+    }
+
+    public OktaResponse getSystemLogs(String domain, String apiKey) throws IOException {
+        // TODO check if client is suitable
+        OkHttpClient client = new OkHttpClient().newBuilder().build();
+        Request request = new Request.Builder()
+                .url(domain)
+                .method("GET", null)
+                .addHeader("Accept", "application/json")
+                .addHeader("Content-Type", "application/json")
+                .addHeader("Authorization", apiKey)
+                .build();
```

Comment:
This `OkHttpClient` is good to use here. 👍 However, please use the `OkHttpClientProvider` to obtain an instance through Guice dependency injection (some global options are set on it, and this pattern allows them to be applied for your usage). Here's an [example](https://github.com/Graylog2/graylog-plugin-enterprise/blob/15d44edb3dc0ec5b3a03b60734882ab310337886/enterprise/src/main/java/org/graylog/plugins/license/LicenseReportPeriodical.java#L78)

Reply:
Thanks @danotorrey for the pointer. also using this link as reference.
[okhttp](https://square.github.io/okhttp/)